Sources/ForageSDK/Foundation/Telemetry/ForageLogger.swift

```diff
@@ -69,29 +68,42 @@ internal protocol ForageLogger {
     func critical(_ message: String, error: Error?, attributes: [String: Encodable]?)
 }
 
+/// Read this [document](https://docs.google.com/document/d/1BU609qv7qSDN-tdNaJP-sAyrgeM6REe9tvcMk5CxN3c/edit#bookmark=id.w44b8kk07o7o) for to learn more about how DatadogLogger works
 internal class DatadogLogger : ForageLogger {
     private static let DD_CLIENT_TOKEN: String = "pub1e4572ba0f5e53df108c333d5ec66c02"
-    private var logger: Logger? = nil
+    private static let DD_SERVICE_NAME: String = "ios-sdk"
+    private static let DD_SDK_INSTANCE_NAME: String = "forage"
+
+    private var logger: LoggerProtocol? = nil
     private var config: ForageLoggerConfig? = nil
-        
+    
     required internal init(_ config: ForageLoggerConfig? = ForageLoggerConfig()) {
+        if isUnitTesting() {
+            // avoid emitting logs when running unit tests
+            return
+        }
```

Comment:
A suggestion for potential improvement at a better time:

The use of `isUnitTesting()` gets the job done and allows us to ship quickly. I think it might be beneficial for us to consider having a more flexible logging mechanism as we continue to scale our application. We could have multiple types of loggers that implement the same interface. For example:

- `DatadogLogger`: For our production and staging logs (already implemented)
- `LocalLogger`: This might be used to log statements locally, especially useful during development.
`NothingLogger`: A logger that essentially does nothing, which can be useful in certain scenarios like unit testing.

Here's a simple example of how this could look inside the ForageSDK:

```swift
// very much abbreviated
public class ForageSDK {
    private init() {
	let useProdLogger = !isUnitTesting()
	let logger = useProdLogger ? DatadogLogger() : NothingLogger() 
    }
}
```
This way, when we're running unit tests, we wouldn't even initialize the DatadogLogger and can avoid adding branching logic to our Loggers since each one would only focus on one use case (e.g. for prod, unit testing etc)

I created [this ticket](https://linear.app/joinforage/issue/FX-490/ios-create-multiple-instances-of-loggers-to-avoid-context-based-if) so we can discuss on Linear and consider it (or not) in the future.

Reply:
Agreed! Would definitely be good to have a factory or something where we can have more control over logs getting emitted to the console, the verbosity and so on.